### PR TITLE
ZEN-29155: Can't create a maintenance window correctly using JSON api.

### PR DESCRIPTION
**Pull request by olst**

### CODE REVIEW

File: Products/Zuul/infos/devicemanagement.py [hunk 1]
```diff
@@ -38,8 +40,26 @@ def startTime(self):
...
+        duration = int(p['durationMinutes']) or int(p['durationHours']) \
+            or int(p['durationDays'])
+        if not duration:
+            p['durationHours'] = 1
+        if p['repeat'] not in self._object.REPEAT:
+            p['repeat'] = self._object.REPEAT[0]
         self._object.manage_editMaintenanceWindow( 
-                                     startDateTime=p['startDateTime'],
+                                     startDateTime=startDateTime,
                                      durationDays=p['durationDays'],
```
Comment (jplouis) [inline Products/Zuul/infos/devicemanagement.py]: Seems like this method is duplicating some default value picking logic defined in the manage_editMaintenanceWindow param and method.  Specifically the logic around startDateTime and startHours etc.... Those values should be passed in only if they are present in the the `p` param.

Why is it default to 1 hour duration if the duration not set?

Seems like an exception should be thrown if an invalid `repeat` param is sent.

**Comment by olst** [comment]
updated

**Comment by olst** [comment]
Please see this videos:
issue: https://youtu.be/-_YnjSHGW58
fix: https://youtu.be/ZDS0QDFA7mg

File: Products/Zuul/infos/devicemanagement.py [hunk 2]
```diff
@@ -39,7 +39,10 @@ def startTime(self):
 
     def updateWindow(self, p):
         self._object.manage_editMaintenanceWindow( 
-                                     startDateTime=p['startDateTime'],
+                                     startDateTime=p.get('startDateTime', None),
+                                     startDate=p.get('startDate', None),
+                                     startHours=p.get('startHours', None),
+                                     startMinutes=p.get('startMinutes', None),
```
Comment (jplouis) [inline Products/Zuul/infos/devicemanagement.py]: If startDateTime, startMinutes and startHours are not in the `p` map and get set to None, the manage_editMaintenanceWindow method will throw an exception.

**Pull request merged**